src/bin/ion/commands/inspect.rs

```diff
@@ -214,24 +241,29 @@ impl<'a, 'b> IonInspector<'a, 'b> {
         reader: &mut SystemReader<AnyEncoding, Input>,
     ) -> Result<()> {
         const TOP_LEVEL_DEPTH: usize = 0;
+        use ExpandedStreamItem::*;
 
         let mut is_first_item = true;
         let mut has_printed_skip_message = false;
         loop {
-            let item = reader.next_item()?;
-            let is_last_item = matches!(item, SystemStreamItem::EndOfStream(_));
-            item.raw_stream_item()
-                .map(|i| self.confirm_encoding_is_supported(i.encoding()))
-                .unwrap_or(Ok(()))?;
-
             if is_first_item {
                 self.write_table_header()?;
             }
 
+            let expr = reader.next_expanded_item()?;
+
+            let maybe_raw_item = expr.raw_item();
+            // If this item is backed by bytes on the wire, make sure we support the encoding.
+            if let Some(raw_item) = maybe_raw_item {
+                self.confirm_encoding_is_supported(raw_item.encoding())?;
+            }
+
+            let is_last_item = matches!(expr, EndOfStream(_));
+
             match self.select_action(
                 TOP_LEVEL_DEPTH,
                 &mut has_printed_skip_message,
-                &item.raw_stream_item(),
+                &maybe_raw_item,
```

Comment:
🗺️ If this stream item isn't ephemeral, see if it's in the requested range of bytes to inspect.